packages/components/src/FilterBar/FilterBar.spec.tsx

```diff
@@ -91,66 +111,182 @@ const FilterBarWrapper = (
 
 describe("<FilterBar />", () => {
   it("renders filters in the provided order", () => {
-    const { getAllByTestId } = render(<FilterBarWrapper />)
+    const { getAllByTestId } = render(
+      <FilterBarWrapper<ValuesSimple> filters={simpleFilters} />
+    )
     const filters = getAllByTestId(TEST_ID__FILTER)
-    expect(filters.length).toBe(4)
+    expect(filters.length).toBe(3)
     expect(filters[0]).toHaveTextContent("Flavour")
-    expect(filters[1]).toHaveTextContent("Topping")
-    expect(filters[2]).toHaveTextContent("Sugar Level")
-    expect(filters[3]).toHaveTextContent("Ice Level")
+    expect(filters[1]).toHaveTextContent("Sugar Level")
+    expect(filters[2]).toHaveTextContent("Ice Level")
   })
 
-  it("allows updating the values via an external event", async () => {
-    const Wrapper = (): JSX.Element => {
-      type ExternalEventValues = {
-        flavour: string
-      }
+  describe("Removable filters", () => {
+    it("shows inactive filters in the Add Filters menu only", async () => {
+      const { getByRole, queryByText } = render(
+        <FilterBarWrapper<ValuesRemovable> filters={filtersRemovable} />
+      )
+      expect(queryByText("Topping")).not.toBeInTheDocument()
 
-      const [values, setValues] = useState<Partial<ExternalEventValues>>({})
-
-      const filters = [
-        {
-          id: "flavour",
-          name: "Flavour",
-          Component: (
-            <FilterBar.Select
-              items={[
-                { value: "honey-milk-tea", label: "Honey Milk Tea" },
-                { value: "lychee-green-tea", label: "Lychee Green Tea" },
-              ]}
-            />
-          ),
-        },
-      ] satisfies Filters<ExternalEventValues>
-
-      return (
-        <div>
-          <FilterBar
-            filters={filters}
-            values={values}
-            onValuesChange={setValues}
-          />
-          <button
-            type="button"
-            onClick={() => setValues({ flavour: "honey-milk-tea" })}
-          >
-            Update Flavour to honey-milk-tea
-          </button>
-        </div>
+      const addFiltersButton = getByRole("button", { name: "Add Filters" })
+      await user.click(addFiltersButton)
+
+      const list = getByRole("list")
+      const menuOption = within(list).getByRole("button", { name: "Topping" })
+
+      await waitFor(() => {
+        expect(menuOption).toBeVisible()
+      })
+    })
+
+    it("shows removable filter as active if there is a default value", () => {
+      const { getByRole } = render(
+        <FilterBarWrapper<ValuesRemovable>
+          filters={filtersRemovable}
+          defaultValues={{ topping: "pearls" }}
+        />
       )
-    }
+      expect(getByRole("button", { name: "Topping : Pearls" })).toBeVisible()
+      expect(
+        getByRole("button", { name: "Remove filter - Topping" })
```

Comment:
Something I picked up in @mcwinter07's PR on Workflows is that he's switching to identifying things by `id` rather than than the labels of steps to avoid having to do i18n string matching when managing the current step.

Perhaps it would be better for us to apply this approach more broadly so that our micro-copy can safely change without the unit tests breaking.

Reply:
We shouldn't be testing by targeting by `id`. Even if we have i18n implemented, I could just ensure these tests are all testing in English. If your translated text changes, then it makes sense for the test to fail. Changing the baked-in copy should be the same as changing a colour on a component. We should be alerted of it.

Reply:
Yeah +1 to selecting things by role and label instead of id, it ensures we're adding accessible names which is really important.